### PR TITLE
Create a goolfc toolchain using CUDA 8.0.44

### PR DESCRIPTION
**Pull request by akesandgren**
Create a goolfc toolchain using CUDA 8.0.44 and the latest versions of the dependencies.

CUDA 8.0.44
GCC 5.4.0-2.26 (Cuda still doesn't support any newer GCC compiler)
FFTW 3.3.5
OpenBLAS 0.2.19/Lapack 3.6.1
OpenMPI 2.0.1
Scalapack 2.0.2

### CODE REVIEW

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-2.0.1-gcccuda-2016.10.0.eb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+toolchain = {'name': 'gcccuda', 'version': '2016.10.0'}
+
+sources = [SOURCELOWER_TAR_GZ]
+
+source_urls = ['http://www.open-mpi.org/software/ompi/v%(version_major_minor)s/downloads',]
+
+#patches = [
+#    'OpenMPI-%(version)s_common-cuda-lib.patch',
+#    'OpenMPI-%(version)s-vt_cupti_events.patch',
+#]
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenMPI/OpenMPI-2.0.1-gcccuda-2016.10.0.eb]: @akesandgren I guess this commented out block can be removed?

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 7 out of 10 (10 easyconfigs in this PR)
node2135.delcatty.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/ca2a77f24c5f753fef171e9acfc930d1 for a full test report.

**Comment by boegel** [comment]
ah, I should fix this in the `ScaLAPACK` easyblock in `develop`...

```
== 2016-10-13 17:08:36,190 easyblock.py:2178 INFO Running method configure_step part of step configure
== 2016-10-13 17:08:36,190 fancylogger.py:269 WARNING DEPRECATED (since v3.0) functionality used: EB_ScaLAPACK.configure_step uses hardcoded list of LAPACK libs; see http://easybuild.readthedocs.org/en/latest/Deprecated-functionality.html for more information
== 2016-10-13 17:08:36,630 build_log.py:163 ERROR DEPRECATED (since v3.0) functionality used: EB_ScaLAPACK.configure_step uses hardcoded list of LAPACK libs; see http://easybuild.readthedocs.org/en/latest/Deprecated-functionality.html for more information (at vsc/utils/fancylogger.py:321 in raiseException)
```

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 8 out of 10 (10 easyconfigs in this PR)
node2656.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/d4eec2014037807a126f905fb0c57bb7 for a full test report.

**Comment by boegel** [comment]
required updates for ScaLAPACK easyblock to make it work again with EasyBuild v3.x implemented in https://github.com/hpcugent/easybuild-easyblocks/pull/1014

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 8 out of 10 (10 easyconfigs in this PR)
node1213.muk.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/35601811c79bb29e640ff1694f9f600f for a full test report.

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-2.0.1-gcccuda-2016.10.0.eb (hunk 1 above)
Comment (akesandgren) [inline easybuild/easyconfigs/o/OpenMPI/OpenMPI-2.0.1-gcccuda-2016.10.0.eb]: Maybe I didn't look at the patches properly. I'll do that and see if they are still needed or not.

Comment (akesandgren) [inline easybuild/easyconfigs/o/OpenMPI/OpenMPI-2.0.1-gcccuda-2016.10.0.eb]: The are not needed and have been removed.

**Comment by akesandgren** [comment]
The avx2 failure on Sandybridge is obvious, but is there a way to easily find out what features are available and add the respective build flags?

**Comment by akesandgren** [comment]
Dropped the enable-avx2 flags for FFTW for the time being.

**Comment by boegel** [comment]
@akesandgren I think the only solution is to create an easyblock for FFTW that checks the processor features of the system it is being installed on, unless instructed otherwise.

See also the discussion on the EasyBuild mailing list recently (https://lists.ugent.be/wws/arc/easybuild/2016-10/msg00000.html).

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 10 out of 10 (10 easyconfigs in this PR)
node1099.muk.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/cd7de0351a8602d4812f8331427ba4e1 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 10 out of 10 (10 easyconfigs in this PR)
node2301.phanpy.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/37554d155ff4d48b3c97783a7dca469b for a full test report.

**Comment by boegel** [comment]
Good to go now, thanks @akesandgren!

**Pull request merged**